odbc/src/api/functions.rs

```diff
@@ -1966,23 +2013,62 @@ pub unsafe extern "C" fn SQLSetConnectAttr(
 /// This is the WChar version of the SQLSetConnectAttr function
 ///
 /// # Safety
-/// Because this is a C-infereface, this is necessarily unsafe
+/// Because this is a C-interface, this is necessarily unsafe
 ///
 #[no_mangle]
 pub unsafe extern "C" fn SQLSetConnectAttrW(
-    _hdbc: HDbc,
-    _attr: ConnectionAttribute,
-    _value: Pointer,
-    _str_length: Integer,
+    connection_handle: HDbc,
+    attribute: ConnectionAttribute,
+    value_ptr: Pointer,
+    str_length: Integer,
 ) -> SqlReturn {
-    unimplemented!()
+    let mut err = None;
+    let conn_handle = MongoHandleRef::from(connection_handle);
+
+    // This scope is introduced to make the RWLock Guard expire before we write
+    // any error values via add_diag_info as RWLock::write is not reentrant on
+    // all operating systems, and the docs say it can panic.
+    let sql_return = {
+        let conn = must_be_valid!((*conn_handle).as_connection());
+        let mut conn_guard = conn.write().unwrap();
+
+        match attribute {
+            ConnectionAttribute::CurrentCatalog => {
```

Comment:
discussed offline -- deferring this work until after the epic since it is not required to get PowerBI working. Followup ticket: https://jira.mongodb.org/browse/SQL-1075

Reply:
This seems like bizarre behavior to me. The current catalog should have no affect on outstanding cursors. Do we really have to support that behavior? It seems fine to me to just not error and have it work.

Reply:
I always try to give them credit for doing things for a reason. I can't find something that makes me want to fight for this behavior.
Maybe if you are in prepare/execute flow and the change happen in between them? Let's you prepare on DB1, you get the resultset metadata, bind your resultset column or just simply prepare your buffer for retrieving the data, switch the catalog to DB2 which has the same table but with a different schema, then you execute and then when retrieving everything is either looking weird/incorrect or failing?
Let talk more about this when we get to SQL-1075.